### DISCUSSION TITLE
Github actions workflow stuck on acquiring runner for 19 hours (and counting) - can't cancel

### DISCUSSION THREAD

**Discussion by qassim** [community/community#44205] [Actions] [answered]
### Select Topic Area

Bug

### Body

Hi,

We have an actions workflow in a private repository, we don't have any self hosted runners assigned and this workflow otherwise ran fine.

In a matrix in one of our workflow jobs, we're stuck with the following message:

```
Requested labels: ubuntu-latest
Job defined at: awazevr/****/.github/workflows/deploy_matrix.yml@refs/heads/main
Reusable workflow chain:
awazevr/****/.github/workflows/main.yml@refs/heads/main (ef505f0049b0782ce3dfa8bb796a13f188597078)
-> awazevr/****/.github/workflows/deploy_matrix.yml@refs/heads/main (ef505f0049b0782ce3dfa8bb796a13f188597078)
Waiting for a runner to pick up this job...
Job is completed before starting.
```

The workflow can't be cancelled (we've tried many times) and because of the concurrency we set up on this job, future workflow runs that are trying to run this job are being blocked.

The main issue we have is that we can't cancel the job. Is there anything we can do from our side to properly force it to cancel?

Thanks,

**Comment by lrozniewicz** [comment]
@qassim Did You, by any chance, find a way to cancel this workflow? I've come across the same issue, and I am unable to cancel it / remove it (GHA GUI, CLI, gh API)

Also, please check if you use `always()` in your workflow. If you do,  you should check this 🧵 
https://github.com/orgs/community/discussions/26303

**Reply by qassim** [reply]
Unfortunately we didn't manage to cancel it - it's still running! We have concurrency locks so we had to rename our concurrency id for that job to get around it.

I do suspect it's our use of always, so thanks for that thread - it's definitely useful!

**Comment by andrewakim** [comment, marked as answer]
:wave: - Just wanted to share that we added ["force a workflow to cancel" to our API](https://docs.github.com/rest/actions/workflow-runs#force-cancel-a-workflow-run).

**Comment by mariusgreuel** [comment]
@andrewakim I also find it suboptimal that although the job did not run, the job is apparently marked as completed ('Job is completed before starting'), and dependent jobs (via `needs`) continue to run. I have a dependent job that collects several build status via `if: ${{ !(contains(needs.*.result, 'failure')) }}` and that now shows **all good**.

For this kind of condition I would expect a big red X.
